tests/js/test-utils.js

```diff
@@ -2,32 +2,51 @@
  * External dependencies
  */
 import { render } from '@testing-library/react';
-import PropTypes from 'prop-types';
+import invariant from 'invariant';
 
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
-import { Fragment } from '@wordpress/element';
-
-const TestProviders = ( { children } ) => {
-	return (
-		<Fragment>
-			{ children }
-		</Fragment>
-	);
-};
-
-TestProviders.defaultProps = {
-	children: undefined,
-};
+import { createTestRegistry } from './utils';
+import { RegistryProvider } from '@wordpress/data';
 
-TestProviders.propTypes = {
-	children: PropTypes.node,
-};
+// Override `@testing-library/react`'s render method with one that includes
+// our data store.
 
-// custom render with the test provider component
-const customRender = ( ui, options ) => {
-	return render( ui, { wrapper: TestProviders, ...options } );
+/**
+ * Renders the given UI into a container to make assertions.
+ *
+ * @see {@link https://testing-library.com/docs/react-testing-library/api#render}
+ *
+ * @param {*} ui Any valid React child element.
+ * @param {Object} options Render options.
+ * @param {Function} options.setupRegistry A function which accepts the registry instance to configure it.
+ * @param {Function} options.registry A specific registry instance to use. Defaults to a fresh test registry with all stores.
+ * @return {Object} An object containing all of {@link https://testing-library.com/docs/react-testing-library/api#render-result}
+ *                 as well as the `registry`.
+ */
+const customRender = ( ui, options = {} ) => {
+	const {
+		setupRegistry = ( r ) => r,
+		registry = createTestRegistry(),
+		...renderOptions
+	} = options;
+
+	invariant( typeof setupRegistry === 'function', 'options.setupRegistry must be a function.' );
+	setupRegistry( registry );
+
+	function Wrapper( { children } ) {
+		return (
+			<RegistryProvider value={ registry }>
+				{ children }
+			</RegistryProvider>
```

Comment:
It was, but I changed it on purpose for better compatibility with the React Testing Library API. The main reason is that if you call `rerender` (one of the functions returned from `render()`) it would invoke the `setupCallback` again, when that should only be called on the initial rendering/setup. `rerender` is more for testing with different props, but there may be cases where we want/need it, and so using `WithTestRegistry` here could lead to bugs.